index.d.ts

```diff
@@ -20,24 +20,36 @@ interface BodyEntry {
 }
 
 export interface MultipartFields {
-    [x: string]: Multipart | Multipart[];
+    [fieldname: string]: Multipart | Multipart[];
 }
 
-export type Multipart<T = true> = T extends true ? MultipartFile : MultipartValue<T>;
+export type Multipart = MultipartFile | MultipartValue;
 
 export interface MultipartFile {
   toBuffer: () => Promise<Buffer>,
   file: BusboyFileStream,
-  filepath: string,
   fieldname: string,
   filename: string,
   encoding: string,
   mimetype: string,
   fields: MultipartFields
 }
 
-export interface MultipartValue<T> {
-  value: T
+export interface SavedMultipartFile extends MultipartFile {
+  /**
+   * Path to the temporary file
+   */
+  filepath: string,
+}
+
+export interface MultipartValue<T = unknown> {
+  value: T;
```

Comment:
The value should be `unknown`, since it might be parsed as JSON based on the content-type provided by the caller. I've left it generic in case someone wants to assert its type.